Lunch/TodaysLunchViewController.swift

```diff
@@ -7,20 +7,150 @@
+                                var annotation = CustomPointAnnotation()
+                                annotation.coordinate = item.placemark.coordinate
+                                annotation.title = item.name
+                                annotation.subtitle = item.phoneNumber
+                                annotationList.append(annotation)
+                                self.mapView.addAnnotation(annotation)
+                            }
+                            self.mapView.showAnnotations(
+                            annotationList, animated: true)
+                        }
+
+                    })
+                }
+            }
+        })
     }
-    
+
+    override func viewDidLayoutSubviews() {
+
+    }
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        selectedPlaceLabel.text = textForLunchLabel
+        selectedVotesLabel.text = "Votes: " + textForVotesLabel
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        let vc = self.navigationController?.topViewController as? ViewController
+        vc?.votes[selectedIndex] = textForVotesLabel.toInt()!
+        vc?.updateTableView()
+    }
+
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
     }
-    
-}
+
+    func mapView(mapView: MKMapView!, viewForAnnotation annotation: MKAnnotation!) -> MKAnnotationView! {
+        if !(annotation is CustomPointAnnotation) {
+            return nil
+        }
+
+        let reuseId = "test"
+
+        var anView = mapView.dequeueReusableAnnotationViewWithIdentifier(reuseId)
+        if anView == nil {
+            anView = MKAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
+            anView.canShowCallout = true
+        }
+        else {
+            anView.annotation = annotation
+        }
+
+        let cpa = annotation as! CustomPointAnnotation
+        anView.image = self.image
+
+        return anView
+    }
+
+    func mapView(mapView: MKMapView!, didSelectAnnotationView view: MKAnnotationView!) {
+        if view.annotation is CustomPointAnnotation {
+            if let phoneNumber = view.annotation.subtitle {
+                let regex = Regex(stringLiteral: "[0-9]*")
+                if !phoneNumber.isEmpty && phoneNumber.match(regex) {
+                    var application = UIApplication.sharedApplication()
+                    var url = NSURL(string: "telprompt://\(phoneNumber!)")
+                    application.openURL(url!)
+                }
+            }
+        }
+    }
+
+    @IBAction func voteNavBarButtonClick(sender: AnyObject) {
+        var query = PFQuery(className: self.classNameKey)
+        query.whereKeyExists(self.placeColumnKey)
+        query.whereKeyExists(self.voteColumnKey)
+        query.findObjectsInBackgroundWithBlock({ (objects: [AnyObject]?, error: NSError?) -> Void in
+            var pfobjects = objects as! [PFObject]
```

Comment:
This force-unwrapping might be alright if the documentation says it'll *always* be PFObjects. Does it?